x-pack/plugins/dataset_quality/server/services/index_stats.ts

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient } from '@kbn/core/server';
+import { rangeQuery } from '@kbn/observability-plugin/server';
+import { DataStreamType } from '../../common/types';
+
+class IndexStatsService {
+  public async getIndexStats(
+    esClient: ElasticsearchClient,
+    type: DataStreamType
+  ): Promise<{
+    doc_count: number;
+    size_in_bytes: number;
+  }> {
+    try {
+      const index = `${type}-*`;
+
+      const indexStats = await esClient.indices.stats({ index });
+      return {
+        doc_count: indexStats._all.primaries?.docs ? indexStats._all.primaries?.docs?.count : 0,
+        size_in_bytes: indexStats._all.primaries?.store
+          ? indexStats._all.primaries?.store.size_in_bytes
+          : 0,
+      };
+    } catch (e) {
+      if (e.statusCode === 404) {
```

Comment:
Do we want to still capture this as an error so we have telemetry (kibana logs) etc when it happens?

Reply:
It's a good point but I don't think it's relevant in this case, we will be using the index name coming out from the dataStreams API, summary is disconnected from the filters, in theory we shouldn't even enter in this branch.